modules/nf-core/dysgu/main.nf

```diff
@@ -0,0 +1,55 @@
+process DYSGU {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/dysgu%3A1.6.2--py310h770aed0_0' :
+        'biocontainers/dysgu:1.6.2--py310h770aed0_0' }"
+
+    input:
+    tuple val(meta), path(input_bam), path(input_bam_index)
+    tuple val(meta2), path(fasta), path(fai)
+
+    output:
+    tuple val(meta), path('*.vcf.gz')       , emit: vcf
+    tuple val(meta), path('*.vcf.gz.tbi')   , emit: tbi
+    path 'versions.yml'                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def args3 = task.ext.args3 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    dysgu run \\
+        $fasta \\
+        --temp-directory ./temp \\
+        $input_bam \\
+        | bgzip ${args2} --threads ${task.cpus} --stdout > ${prefix}.vcf.gz
+    tabix ${args3} ${prefix}.vcf.gz
```

Comment:
I think we have to create the tmp directory before using it.
```suggestion
    """
    mkdir temp
    dysgu run \\
        $fasta \\
        --temp-directory ./temp \\
        $input_bam \\
        | bgzip ${args2} --threads ${task.cpus} --stdout > ${prefix}.vcf.gz
    tabix ${args3} ${prefix}.vcf.gz
```

Reply:
> I think we have to create the tmp directory before using it.

Reply:
I am wondering if we can just use the current dir as temp. So just 
`--temp-directory .`

Reply:
But if that does not work try the other way.

Reply:
I have tried the both ways and in the end this `--temp-directory .` didn't work so I tried the other way and it worked and finished with some errors.

> 2024-06-24 10:01:33,128 [INFO   ]  [dysgu-run] Version: 1.6.2
> 2024-06-24 10:01:33,128 [INFO   ]  run genome.fasta temp_dysgu test.paired_end.recalibrated.sorted.cram
> 2024-06-24 10:01:33,128 [INFO   ]  Destination: temp_dysgu
> [W::cram_get_ref] Reference file given, but ref 'chr21' not present
> [W::cram_populate_ref] Creating reference cache directory //.cache/hts-ref
> This may become large; see the samtools(1) manual page REF_CACHE discussion
> //.cache/hts-ref/97/4d/c7aec0b755b19f031418fdedf293.tmp_31_1_2002320682: No such file or directory
> 2024-06-24 10:02:01,288 [INFO   ]  dysgu fetch test.paired_end.recalibrated.sorted.cram written to temp_dysgu/test.paired_end.recalibrated.sorted.dysgu_reads.bam, n=252, time=0:00:28 h:m:s
> 2024-06-24 10:02:01,288 [INFO   ]  Input file is: temp_dysgu/test.paired_end.recalibrated.sorted.dysgu_reads.bam
> 2024-06-24 10:02:01,289 [INFO   ]  Sample name: normal
> 2024-06-24 10:02:01,289 [INFO   ]  Writing vcf to stdout
> 2024-06-24 10:02:01,289 [INFO   ]  Running pipeline
> [W::cram_get_ref] Reference file given, but ref 'chr21' not present
> [W::cram_populate_ref] Creating reference cache directory //.cache/hts-ref
> This may become large; see the samtools(1) manual page REF_CACHE discussion
> //.cache/hts-ref/97/4d/c7aec0b755b19f031418fdedf293.tmp_31_1_2000984014: No such file or directory
> 2024-06-24 10:02:56,170 [INFO   ]  Calculating insert size. Removed 0 outliers with insert size >= 1359.0
> 2024-06-24 10:02:56,180 [INFO   ]  Inferred read length 100.0, insert median 351, insert stdev 145
> 2024-06-24 10:02:56,181 [INFO   ]  Max clustering dist 1076
> 2024-06-24 10:02:56,181 [INFO   ]  Building graph with clustering 1076 bp
> 2024-06-24 10:02:56,183 [INFO   ]  Total input reads 252
> 2024-06-24 10:02:56,183 [INFO   ]  Graph constructed
> 2024-06-24 10:02:56,183 [INFO   ]  Minimum support 3
> 2024-06-24 10:02:56,196 [CRITICA]  No events found
> 2024-06-24 10:02:56,198 [INFO   ]  dysgu run test.paired_end.recalibrated.sorted.cram complete, time=0:01:23 h:m:s
[command.log](https://github.com/user-attachments/files/15953760/command.log)

Reply:
How did you define the temp-dir now?

Reply:
Hi @famosab I have give a temp directory as a input in command line of dysgu example: 
```
dysgu run \\
        $fasta \\
        temp_dysgu \\  ###I have added assign this file as temp directory, and dysgu creates all the temp files inside it.
        $input_bam \\
        | bgzip ${args2} --threads ${task.cpus} --stdout > ${prefix}.vcf.gz
    tabix ${args3} ${prefix}.vcf.gz
```

Reply:
ok I think that probably wont work with the module because you have to create the temp from outside?

Reply:
> This will first run the fetch command that will create a temporary bam file plus other analysis files in the directory temp_dir. These temporary files are then analysed using the call program. To make use of multiprocessing, set the "-p" parameter:

Ok I thionk we do not have to create any temp dir, my bad. Sorry! But good to know that this commadn runs dysgu fetch. We should however set the `-p`

Reply:
> ok I think that probably wont work with the module because you have to create the temp from outside?

Upon re-reading the dysgu GitHub page, I believe I misunderstood the concept of "working_directory" as "temp_directory." I think specifying a working directory instead of a temp directory might resolve the issue and allow dysgu to function effectively as a module, similar to "manta."
PS: [`Also a working directory must be provided to store temporary files`]

Reply:
So normally we can use `.` as the tmp directory for tools that needs it. It will point to whatever nextflow has configured as tmp for the particular setup. (see i.e. the GATK modules)

Reply:
> > This will first run the fetch command that will create a temporary bam file plus other analysis files in the directory temp_dir. These temporary files are then analysed using the call program. To make use of multiprocessing, set the "-p" parameter:
> 
> Ok I thionk we do not have to create any temp dir, my bad. Sorry! But good to know that this commadn runs dysgu fetch. We should however set the `-p`

maybe we can try both options? 
1. `dysgu run -p4 reference.fa temp_dir input.bam > svs.vcf`
2. we can try adding working_directory [if needed]

Reply:
> So normally we can use `.` as the tmp directory for tools that needs it. It will point to whatever nextflow has configured as tmp for the particular setup. (see i.e. the GATK modules)

Hi @FriederikeHanssen we used `.`as temp directory in earlier test runs but unfortunately it returns with `Error: No such option: --temp-directory`

Reply:
@poddarharsh15 I think thats because the flag is not used by dysgu not because of the path being not possible to use